syntax = "proto3";

package store;

import "google/protobuf/empty.proto";  // Importa a definição padrão de Empty

message Price {
    int32 value = 1;
}

message PurchaseRequest {
    int32 order_id = 1;
}

message PurchaseResponse {
    int32 status = 1;
}

message StoreShutdownResponse {
    int32 seller_balance = 1;
    int32 wallet_shutdown_status = 2;
}

service Store {
    rpc GetPrice (google.protobuf.Empty) returns (Price);
    rpc Purchase (PurchaseRequest) returns (PurchaseResponse);
    rpc Shutdown (google.protobuf.Empty) returns (StoreShutdownResponse);
}

